libwebauthn/src/transport/channel.rs

```diff
@@ -30,3 +31,45 @@ pub trait Channel: Send + Sync + Display {
     async fn cbor_send(&mut self, request: &CborRequest, timeout: Duration) -> Result<(), Error>;
     async fn cbor_recv(&mut self, timeout: Duration) -> Result<CborResponse, Error>;
 }
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct Ctap2AuthTokenIdentifier {
+    pin_uv_auth_protocol: Ctap2PinUvAuthProtocol,
+    permissions: ClientPinRequestPermissions,
+    permissions_rpid: Option<String>,
+}
+
+impl Ctap2AuthTokenIdentifier {
+    pub fn new(
+        pin_uv_auth_protocol: Ctap2PinUvAuthProtocol,
+        permissions: ClientPinRequestPermissions,
+        permissions_rpid: Option<&str>,
+    ) -> Self {
+        Self {
+            pin_uv_auth_protocol,
+            permissions,
+            permissions_rpid: permissions_rpid.map(str::to_string),
+        }
+    }
+
+    pub fn is_satisfied_by(&self, other: &Ctap2AuthTokenIdentifier) -> bool {
```

Comment:
I'm not a native English speaker, but I find this terminology (identifier <- is satisfied by <- `other` identifier) a bit confusing.

I wonder if it could be simplified further, e.g.:
* identifier <- is_subset_of <- `other` identifier
* permissions -> allows -> permissions `request`

Reply:
Yes, I was not at all happy with my choice of words here, either. 

And, as you pointed out below, I got it backwards as well. I should have used `satisfies()` instead of `is_satisfied_by()`.

`allows()` sounds reasonable. Although, right now I'm also thinking about `contains()`:
```
if stored_permission.contains(requested_permission) {
    return stored_token;
}
```

I'd be happy with both. WDYT?